hasura/graphql-engine
# added export data option on data module (close #1438 #5158)

### Description
This PR adds a feature that allows Hasura console users to export data to JSON or CSV file. 

### Changelog


### Affected components

### Related Issues
close #1438 
close #5158

### Solution and Design
A query with the filter is used to hit hasura backend, which will not have `limit` / `offset` so all the entries with the applied filter can be retrieved from the server. this data is hence processed to the required format (JSON/CSV).  

### Steps to test and verify
![image](https://user-images.githubusercontent.com/8408875/88142265-d6983b80-cc12-11ea-900e-c51564f41885.png)

* User can now go to data module and see the export button as shown
* By clicking the export option, CSV/JSON download should start with the correct data.

### Limitations, known bugs & workarounds
* since we are fetching all the data without a limit, large tables might have a performance issue.

### Server checklist


#### Catalog upgrade

#### Metadata
Does this PR add a new Metadata feature?

#### GraphQL

#### Breaking changes
